### PR TITLE
Add OpenShift documentation

### PR DESCRIPTION
**Pull request by npepinpe**
**Description**

This PR adds documentation for self-managed platform deployments on OpenShift. I followed the existing pattern of having a general Kubernetes guide (without Helm) and a specific Helm chart deployment guide.

Both guides have some overlap, I guess, but I think generally the information is useful. I'm happy to also merge them and focus on a single Helm specific guide, ignoring non-Helm deployments on OpenShift :shrug: 

I also didn't want to link too much to the values and other files in the repository, as I wanted the documentation here to be as self-contained as possible. I still linked to the repository as a quick-start for impatient users, while taking more time to explain here in these docs. Let me know what you think.

### CODE REVIEW

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift.md:73]: It's hard to understand this section.

File: docs/self-managed/platform-deployment/kubernetes-helm.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/kubernetes-helm.md:89]: Not sure if it's a convention, but this title (like other titles here) is redundant.
We are already on the Helm page under `How to install Camunda 8 using Helm` I believe that there is no need to have the full line again here.

something like this should be fine:
> Install on OpenShift

File: docs/self-managed/platform-deployment/openshift-helm.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift-helm.md:53]: For better usability and accessibility, it's [recommended](https://webaccess.berkeley.edu/ask-pecan/click-here) to avoid using `here` for hyperlinks and put the link under a descriptive name.

e.g.
> also a predefined [values file](https://github.com/camunda/camunda-platform-helm/blob/main/openshift/values.yaml) is available
> also you can use this predefined [values.yaml](https://github.com/camunda/camunda-platform-helm/blob/main/openshift/values.yaml) file

File: docs/self-managed/platform-deployment/openshift-helm.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift-helm.md:140]: the same point about using `here` for hyperlinks.

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift.md:9]: the same point about using `here` for hyperlinks.

File: docs/self-managed/platform-deployment/openshift-helm.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift-helm.md:192]: I think we need `warning` here.

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [inline docs/self-managed/platform-deployment/openshift.md:16]: @npepinpe what do you think about putting an `H2` title called `Caveats` where we can use it as a unified way across platforms (also for EKS and AKS later).

so the structure would be like this:

```
## Deployment 
## Caveats
### Security Context Constraints (SCC)
#### Permissive SCC
#### Non-root SCC
#### Restrictive SCC
```

**Comment by pepopowitz** [comment]
Hi friends!! I want to make sure you're aware of #1118 -- a PR that I plan to merge in the next day. 

It makes some fundamental changes to where our docs are sourced: 
1. 8.0 docs will now be sourced from `/versioned_docs/version-8.0` instead of `/docs`
2. Anything in `/docs` will be presented as the next/unreleased version

### Action required

I'm not sure if this PR is intended to update 8.0 or 8.1 docs.

* If it is 8.0, one of these things need to happen: 

  1. We need to merge this before I merge #1118. This is my favorite option if the PR is ready, because it means no extra work for either of us 😄 However, I am planning to merge #1118 in the next 24 hours. 
  2. After I merge #1118, you'll need to update this PR to make changes in the `/versioned_docs/version-8.0` folder instead of `/docs`. 

* If this PR is intended for future 8.1 docs, we don't need to do anything! After #1118 merges, this PR _will_ be pointing at the unreleased 8.1 docs, and you can merge it at your convenience. 

---

If you are able to get this one merged in the next day, great! If it's close to done and you want me to wait before merging #1118, I can be convinced. 

If it's not ready yet, I'm happy to work with you to get it re-opened (or modified) to point at the correct source docs.

Thanks! If you have questions or concerns, let me know.

**Comment by npepinpe** [comment]
I have no strong opinion here. This fits in both 8.0 and 8.1, and is not version specific. It is new, however, so possibly users will not look in older version for it. I'm fine with either, really.

File: docs/self-managed/platform-deployment/kubernetes-helm.md
Comment (npepinpe) [reply docs/self-managed/platform-deployment/kubernetes-helm.md:89]: As you pointed out, I was just following what seemed to be the convention :smile: I'm happy to apply your suggestion

File: docs/self-managed/platform-deployment/openshift-helm.md
Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift-helm.md:192]: :+1: I will also move it a bit higher in the section. I think as the last note it's easy to miss.

File: docs/self-managed/platform-deployment/openshift.md
Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift.md:16]: I think the general idea is good, but I'm not sure how to make it flow here. Since the deployment section specifies that, well, how you deploy hinges entirely on which SCC you choose, it feels strange to add the `Caveat` section, which is essentially a warning section.

File: docs/self-managed/platform-deployment/openshift.md
Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift.md:73]: Can you elaborate? Is it the phrasing, or should I define some nomenclature, or is it throwing in concepts I haven't explained yet, etc.?

Comment (aabouzaid) [reply docs/self-managed/platform-deployment/openshift.md:73]: The sentence is hard to read/understand :sweat_smile: 

> If you are using Helm charts to deploy non-Camunda applications (e.g. Keycloak, Elasticsearch, etc.),

Do you mean within Camunda Platform charts or if someone installed these components externally? (outside C8SM charts) 

It's not clear because we actually use external charts for non-Camunda apps. (e.g. Keycloak, Elasticsearch, etc.)

> refer to their own Helm charts on how to configure it for OpenShift.

The word `refer` here means `follow`? If yes, then I'd suggest to use `follow`
because when I read `refer` I think about editing the C8SM chart dependencies and replace the external charts there (e.g. if someone want to use homemade ES chart instead of the official one).

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/openshift.md:16]: My idea is that all K8s distributions (OpenShift, EKS, AKS, etc.) should work the same way ... so we have a single guide to install the charts on K8s. However, if there is anything else that should be done I'd consider it as `odd` and needs user attention.

Maybe `Caveat` is not the best word for that ... maybe "Extra steps" or something like that.

In all cases, I don't have a strong opinion about having that section but the main point here is to set the structure under some H2 section.

So this (without the `Caveat`) still stands:
```
## Deployment 
### Security Context Constraints (SCC)
#### Permissive SCC
#### Non-root SCC
#### Restrictive SCC
```

Which simply reads as "What are the things you as a user need to handle to deploy C8SM on OpenShift".

Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift.md:16]: We could rephrase `Deployment` to something more meaningful. `Required changes`? Maybe just `Configuration`?

Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift.md:16]: I'll go with simply nesting the other headers for now.

File: docs/self-managed/platform-deployment/openshift.md
Comment (npepinpe) [reply docs/self-managed/platform-deployment/openshift.md:73]: I'm thinking of just dropping this block. I'm not sure it's all that useful, actually: should we document how to deploy Elasticsearch standalone on OpenShift, _without our Helm charts_? I'm just not sure it's valuable. So I'd opt to simply remove it.

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/openshift.md]: "Configuration" sounds great :ok_hand:

File: docs/self-managed/platform-deployment/openshift.md
Comment (aabouzaid) [reply docs/self-managed/platform-deployment/openshift.md]: I agree to remove it :+1:

**Review by aabouzaid — APPROVED** [review]
Great job @npepinpe :rocket:

**Comment by npepinpe** [comment]
@aabouzaid I rebased, and apparently that means you need to re-approve, sorry, I didn't know :see_no_evil:

**Review by aabouzaid — APPROVED** [review]

**Comment by aabouzaid** [comment]
@npepinpe no worries, all good :bow:

**Comment by akeller** [comment]
Hi folks, it looks like this got merged in without a docs review. Before merging to `main` or a release branch, please tag @christinaausley or me to review.

Merging to `main` does not automatically publish these docs (yet!). Is this something you would like to see published and available live?

**Comment by npepinpe** [comment]
Hi, you're completely right and I just :100: forgot to do this :see_no_evil: I'm happy to revert the commit and open a new PR for someone to review if you'd like. We would like to see this live as soon as possible to get feedback from customers.

**Comment by akeller** [comment]
@christinaausley how would you like to proceed to review this? Do you want to open a PR yourself with any changes to this content or have @npepinpe revert the commit?

**Comment by akeller** [comment]
Actually, given that @pepopowitz wants to merge #1118 today, @christinaausley please open a PR with your changes on this content. Coordinate with @pepopowitz whether that is going to happen before or after his merge.

**Comment by aabouzaid** [comment]
@akeller I'd suggest enabling Github [code owners](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to make sure the PRs are always reviewed by certain people.

**Comment by akeller** [comment]
@aabouzaid PRs for each team may be opened and completed quickly or are in a draft state (and may or may not be a draft PR) we ask teams to add the docs team for review when the PR is ready. See more info here on the [Docs Review at the Implementation phase](https://confluence.camunda.com/x/pQFKBg). 

I like the idea of code owners, but I want to adjust to the PDP first. Adding reviewers automatically would create a lot of noise if the PR isn't actually ready for review.

**Comment by aabouzaid** [comment]
@akeller totally makes sense :+1: 
In that case, then I'd suggest using [pull request template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository).

It's not necessary to be done now but just to keep in mind.

I'm just saying that because of my bad memory probably I'm the next person who will forget that if it's not explicitly mentioned in the workflow :sweat_smile:

**Comment by christinaausley** [comment]
[PR](https://github.com/camunda/camunda-platform-docs/pull/1137) is ready. There was one sentence I was a bit confused on @npepinpe so I tagged you in a comment. Can launch another PR for this minor fix if you don't see this today.

**Comment by npepinpe** [comment]
Thanks for the review, and apologies again for mergingvtoo early 🙏

**Comment by christinaausley** [comment]
No problem @npepinpe 👍

**Pull request merged**
